iosApp/Survey/Sources/Presentation/Modules/SurveyDetail/SurveyDetailView.swift

```diff
@@ -103,37 +103,60 @@ struct SurveyDetailView: View {
         .frame(maxWidth: .infinity, alignment: .leading)
     }
 
-    var nextButton: some View {
+    var nextButtonArea: some View {
         HStack {
             Spacer()
             if dataSource.isShowingTitle {
-                Button {
-                    dataSource.didPressNext()
-                } label: {
-                    Text(String.localizeId.survey_detail_start_button())
-                        .primaryButton()
-                }
-                .padding(.bottom, .mediumPadding)
-                .accessibility(.surveyDetail(.startButton))
+                startButton
+            } else if (dataSource.viewState.surveyDetail?.questions.count ?? 0) == questionIndex + 1 {
```

Comment:
What do you think about creating another published var in `dataSource` for return questions.count with `private set`? This can help reduce the need for lengthy computation inside an else if block here and improve readability. 💭

Reply:
This `nextButton: View` will be converted later to boolean type from datasource and the logic will move to KMM ViewModel in the integration task 🙏